src/scripts/main.js

```diff
@@ -1,11 +1,20 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(`${eventName}`, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
 }
 
+const createMessage = document.createElement('div');
```

Comment:
The `createMessage` element is created globally but is only used within the `printMessage` function. Consider moving its creation inside the `printMessage` function to encapsulate its usage and avoid potential side effects.